File: api/Dto/User/Friend/FriendRequest/FriendRequestDto.cs
using System.Runtime.Serialization;
using api.Dto.User.Friend;

namespace api.Dto.User
{
  [DataContract]
  public class FriendRequestDto
  {
    [DataMember(Name = "id")]
    public int Id { get; set; }
    [DataMember(Name = "userSentId")]
    public int UserSentId { get; set; }

    [DataMember(Name = "userReceivedId")]
    public int UserReceivedId { get; set; }
    [DataMember(Name = "userSent")]
    public UserDto? UserSent { get; set; }

    [DataMember(Name = "userReceived")]
    public UserDto? UserReceived { get; set; }

    [DataMember(Name = "status")]
    public FriendRequestStatus Status { get; set; }
    [DataMember(Name = "created")]
    public DateTime Created { get; set; }
  }
}

File: src/app/api/User/Friend/friend request/friend-request.dto.ts
import { User } from "src/app/api/User/user.dto";
import { FriendRequestStatus } from "./friend-request-status.enum";

export interface FriendRequest {
    id: number;
    userSentId: number;
    userReceivedId: number;
    userSent?: User;
    userReceived?: User;
    status: FriendRequestStatus;
    created: Date;
}